### PR TITLE
Create `ComponentChecklist` component

### PR DESCRIPTION
**Pull request by colebemis**
Adds a `ComponentChecklist` component in the `/docs` directory that will enable us to document the maturity of our components.

```
## Component checklist

<ComponentChecklist
  items={{
    propsDocumented: true,
    noUnnecessaryDeps: true,
    adaptsToThemes: true,
    adaptsToScreenSizes: true,
    fullTestCoverage: true,
    usedInProduction: true,
    usageExamplesDocumented: true,
    designReviewed: null,
    a11yReviewed: null,
    stableApi: false,
    addressedApiFeedback: false,
    hasDesignGuidelines: null,
    hasFigmaComponent: null
  }}
/>
```

CleanShot 2021-11-18 at 17 16 07

## Known issues

Here are list of known issues with `ComponentChecklist` that I think are out of scope for this PR:

* **The component is not written in TypeScript.** We don't have Gatsby configured to work with TypeScript yet so I wrote the component in vanilla JavaScript. Updating our Gatsby configuration to work with TypeScript felt out of scope for this PR.
* **The `ComponentChecklist` component needs to be imported in every markdown file that uses it.** I plan to make a change in Doctocat that will allow us to make `ComponentChecklist` globally available in all markdown files. Until I make that change, `ComponentChecklist` will need to be explicitly imported.
* **`ComponentChecklist` items are hard to guess.** You'll need to look at the definition of `ComponentChecklist` to know which checklist items to pass. I imagine most people will use `ComponentChecklist` by copying and pasting from other documentation pages, so I don't think this is a dealbreaker
* **You need to add the checklist heading in markdown.** The heading (i.e. `## Component checklist`) needs to be written directly in the `.md` or `.mdx` file. The heading can't be included in the `ComponentChecklist` component because it won't be listed in the table of contents.

## Open questions

- What should the heading of this checklist be? I'm using `Component checklist` for now but I'm open to other ideas.
- Is the `n/a` state for a checklist item visually clear? I'm using the same icon that we use to represent a "skipped" GitHub action.

## Next steps

- [ ] Render this checklist on every component documentation page

## Notes

This is a departure from my original implementation plan for component checklists. I wrote about why changed my implementation plan [here](https://github.com/github/primer/issues/339#issuecomment-972370931).

### CODE REVIEW

**Review by siddharthkp — APPROVED** [review]
This looks good! No changes needed here, suggested changes in doctocat.

**Comment by colebemis** [comment]
@siddharthkp @jfuchs Is this `n/a` state more clear?

CleanShot 2021-11-19 at 08 46 15

**Comment by siddharthkp** [comment]
> Is this n/a state more clear?

It's very clear. Bonus: now we make it pretty too 😋

**Comment by colebemis** [comment]
Better?
CleanShot 2021-11-19 at 08 59 30

**Comment by siddharthkp** [comment]
Love it!

**Review by rezrah — APPROVED** [review]
Very cool @colebemis. As a name suggestion, how about "Component status", as provides more detailed explanation - and is directly related to - the status at the top of the page?

File: docs/src/component-checklist.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import {Box, StyledOcticon, Link, Text} from '@primer/components'
+import {H3} from '@primer/gatsby-theme-doctocat/src/components/heading'
+import {CheckCircleFillIcon, CircleIcon, SkipIcon} from '@primer/octicons-react'
+import React from 'react'
+
+/** Render component maturity checklist in documentation pages */
```
Comment (colebemis) [inline docs/src/component-checklist.js:6]:
```suggestion
/** Render component status checklist in documentation pages */
```

File: docs/content/Box.mdx [hunk 2]
```diff
@@ -1,10 +1,12 @@
 ---
 title: Box
+status: Alpha
```
Comment (colebemis) [inline docs/content/Box.mdx:3]:
```suggestion
status: Beta
```

**Pull request merged**